src/niivue.js

```diff
@@ -727,8 +677,20 @@ Niivue.prototype.refreshLayers = function(overlayItem, layer, numLayers) {
 		this.gl.texStorage3D(this.gl.TEXTURE_3D, 6, this.gl.R16UI, hdr.dims[1], hdr.dims[2], hdr.dims[3]);
 		this.gl.texSubImage3D(this.gl.TEXTURE_3D, 0, 0, 0, 0, hdr.dims[1], hdr.dims[2], hdr.dims[3], this.gl.RED_INTEGER, this.gl.UNSIGNED_SHORT, imgRaw);
 	}
-	if (overlayItem.global_min === undefined) //only once, first time volume is loaded
-		this.calMinMax(overlayItem, imgRaw)
+	if (overlayItem.global_min === undefined) {//only once, first time volume is loaded
+		if(!wasmLoaded) {
+			wasmPromise.then((WasmModule) => {
```

Comment:
I think this promise is resolving after `refreshLayers` has finished executing. `overlayItem.cal_min` and `overlayItem.cal_max` properties don't exist until `this.calMinMax` executes. So the canvas appears blank because it doesn't know what values to use. If you hardcode:

```
overlayItem.cal_min = 40
overlayItem.cal_max = 80
```

before the `wasmPromise` you can see that `cal_min` and `cal_max` are set synchronously and the background image is then correctly drawn on the canvas. 

I think you can prob keep your `wasmPromise` block of code, but wrap it in an `async` function and then `await` it here. 

I think this explains my intermittent loading in Chrome (and broken image loading in firefox and safari)

Reply:
Unfortunately this function is not async.

Reply:
right. I believe you could wrap `wasmPromise` in an async function and then call it synchronously when needed by using `await`.

something like:

```
myfunc = async function (overlayItem, imgRaw) {
    wasmPromise.then((WasmModule) => {
        wasmLoaded = true;
        Module = WasmModule;
        Niivue.prototype.robust_range = Module.cwrap('robust_range', 'number', ['number', 'number']);
       this.calMinMax(overlayItem, imgRaw);
    })
}

// then call it and wait for it
await myfunc(overlayItem, imgRaw)
```
What do you think about adding that `robust_range` method to the Niivue class elsewhere? I guess in that case it would need to init the wasm much earlier? The robust range method is essentially never defined until `refreshLayers` is called in this section of code right? It's sort of hidden. Maybe we can have a pure javascript robust range, so it's always available in the Niivue class, and just override it with the wasm version if wasm is available? 

Just brainstorming.

Treat that code above as pseudo code, I'm sure it could be made a bit better.